atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/sweep/GetCandidateCellsForSweepingIterator.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra.sweep;
+
+import java.util.List;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.Iterables;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweeping;
+import com.palantir.atlasdb.keyvalue.api.CandidateCellForSweepingRequest;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
+import com.palantir.atlasdb.keyvalue.api.RangeRequests;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.cassandra.CqlExecutor;
+import com.palantir.common.base.ClosableIterator;
+
+public class GetCandidateCellsForSweepingIterator extends AbstractIterator<List<CandidateCellForSweeping>>
+        implements ClosableIterator<List<CandidateCellForSweeping>> {
+
+    private final KeyValueService kvs;
+    private final CqlExecutor cqlExecutor;
+    private final TableReference table;
+    private final CandidateCellForSweepingRequest request;
+
+    byte[] nextStartRow;
+
+    public GetCandidateCellsForSweepingIterator(
+            KeyValueService kvs,
+            CqlExecutor cqlExecutor, TableReference table,
+            CandidateCellForSweepingRequest request) {
+        this.kvs = kvs;
```

Comment:
We take the whole KVS, only to pass it via a few intermediate classes, eventually into a helper class that uses one method of it. Is it feasible/cleaner to pass a lambda instead?

Reply:
Yeah probably worth passing extracting an interface for just the functionality we need. Part of the problem is that the `CassandraKvsImpl` is huge, and not broken down into any components.